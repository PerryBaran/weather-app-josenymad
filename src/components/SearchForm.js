import React from "react";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <label htmlFor="search-location" className="search-form__label">
        <input
          type="text"
          id="search-location"
          placeholder="City"
          name="search-location"
          className="search-form__input"
          onChange={handleInputChange}
          value={searchText}
        />
      </label>
      <input
        type="submit"
        value="Search"
        className="search-form__submit"
        onClick={onSubmit}
      />
    </div>
  );
}

export default SearchForm;
