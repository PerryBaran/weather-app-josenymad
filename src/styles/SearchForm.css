.search-form {
    margin-top: 1vh;
    margin-bottom: 3vh;
}

.search-form__submit {
    margin-left: 1vw;
}