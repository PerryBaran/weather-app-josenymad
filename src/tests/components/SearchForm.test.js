import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    searchText: "Test searchText",
    setSearchText: "Test setSearchText",
    onSubmit: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("has a submit button", () => {
    const { getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(getByRole("button")).toBeInTheDocument();
  });

  it("tests the submit button has 'Search' value", () => {
    const { getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    expect(getByRole("button")).toHaveValue("Search");
  });

  it("Search button calls passed down function", () => {
    const { getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );

    fireEvent.click(getByRole("button"));
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
