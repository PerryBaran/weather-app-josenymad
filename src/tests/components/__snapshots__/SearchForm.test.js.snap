// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-form__input"
      placeholder="City"
      type="text"
      value="Test searchText"
    />
    <input
      class="search-form__submit"
      type="submit"
      value="Search"
    />
  </div>
</DocumentFragment>
`;
